.github/workflows/proposals.yml

```diff
@@ -0,0 +1,34 @@
+name: Validate proposals with trestle
+on:
+  pull_request:
+    branches:
+      - main
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Don't mess with line endings
+      run: |
+        git config --global core.autocrlf false
+
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Set up Python
```

Comment:
Nit: Thoughts on using the built in [caching feature](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages) for pip instead of `actions/cache`?